src/core/data/flexible_type/flexible_type.hpp

```diff
@@ -2630,6 +2636,39 @@ inline FLEX_ALWAYS_INLINE_FLATTEN EXT_ENABLE_IF(bool) operator<=(const T& other,
  */
 template <typename T, typename S>
 inline FLEX_ALWAYS_INLINE_FLATTEN EXT_ENABLE_IF(bool) operator>=(const T& other, const S& f) { return (f <= other); }
+
+
+bool FLEX_ALWAYS_INLINE_FLATTEN flexible_type::is_na() const {
```

Comment:
Why does this need to be in the header file? Isn't it sufficient to put it in the .cpp and let the linker do the inlining across translation units?

I'm also curious about the performance difference with and without inlining this function. What does the difference look like for various scenarios?

Reply:
We don't have link time optimization turned on, so the linker doesn't do anything. 

I haven't benchmarked it, but since it's called in an inner loop, is extremely simple, and can easily benefit from type-aware optimizations, so I felt inlining it would be a good optimization to do.

Reply:
Oh, right. We should probably turn on link time optimization (or at least try it out) and benchmark the results.

I'm skeptical of any so-called optimization that doesn't come with a before & after performance profile, taking into account our real user scenarios. I think we can potentially add a lot of complexity, increased build time, etc. in the name of optimization without actually making any measurable improvements if we aren't being data-driven here.

Reply:
Fair, agreed.  We should chat with @Jarvi-Izana on Monday about a good benchmark suite setup.